### PR TITLE
drivers/isl29125: apply unified params definition scheme and new i2c API

### PR DESCRIPTION
**Pull request by MichelRottleuthner**
### Contribution description
Analog to #8688 this updates the parameter scheme (#7519) and applies the new i2c API to the isl29125 driver.

tested with sparkfun isl29125 with nucleo-l073

### Issues/PRs references
#6577

### CODE REVIEW

**Comment by dylad** [comment]
This looks good. I'm supposed to receive custom board with this sensor very soon so I'll give it a try asap unless someone is faster than me.

File: drivers/isl29125/include/isl29125_params.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Set default configuration parameters
+ * @{
+ */
+#ifndef ISL29125_PARAM_I2C
+#define ISL29125_PARAM_I2C              I2C_DEV(0)
```
Comment (MrKevinWeiss) [inline drivers/isl29125/include/isl29125_params.h]: Probably should == TEST_ISL29125_I2C

File: drivers/isl29125/include/isl29125_params.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#endif
+#ifndef ISL29125_PARAM_GPIO
+#define ISL29125_PARAM_GPIO             (GPIO_PIN(0, 0))
```
Comment (MrKevinWeiss) [inline drivers/isl29125/include/isl29125_params.h]: Same here TEST_ISL29125_IRQ_PIN

File: drivers/isl29125/include/isl29125_params.h (hunk 1 above)
Comment (MichelRottleuthner) [reply drivers/isl29125/include/isl29125_params.h]: I removed the parameters from the tests Makefile so there is only the one definition in this file, which can still be overwritten.

File: drivers/isl29125/include/isl29125_params.h (hunk 2 above)
Comment (MichelRottleuthner) [reply drivers/isl29125/include/isl29125_params.h]: addressed: same as above

**Comment by MrKevinWeiss** [comment]
ACK, seems good to me and appears to work!

**Pull request merged**